extensions/notification/queueProcessor/QueueProcessor.js

```diff
@@ -116,17 +116,20 @@ class QueueProcessor extends EventEmitter {
      * @param {boolean} [options.disableConsumer] - true to disable
      *   startup of consumer (for testing: one has to call
      *   processQueueEntry() explicitly)
+     * @param {function} done callback
      * @return {undefined}
      */
-    start(options) {
+    start(options, done) {
         async.series([
             next => this._setupZookeeper(next),
             next => this._setupNotificationConfigManager(next),
             next => this._setupDestination(this.destinationConfig.type, next),
-            next => this._destination.init(() => {
+            // if connection to destination fails, process will stop & restart
+            next => this._destination.init(next),
```

Comment:
About this it looks like on zenko it could crash with a "callback already called" if destination Kafka is not up on startup